lib/chibi/json.c

```diff
@@ -80,8 +81,12 @@ sexp parse_json_string (sexp ctx, sexp self, sexp str, const char* s, int* i, co
         from = to+1;
         break;
       case 'u':
-        strncpy(utf_tmp, s+to+1, 5);
-        tmp = sexp_make_string(ctx, sexp_make_fixnum(1), sexp_make_character(strtoll(utf_tmp, NULL, 16)));
+        strncpy(utf_tmp, s+to+1, 4);
+        tmp = sexp_make_string(ctx, sexp_make_fixnum(1), sexp_make_character(strtoll(utf_tmp, &utf_cursor, 16)));
```

Comment:
I guess strtol is sufficient?  This will never need more than 21 bits.

Reply:
I'm planning full utf support and in json that's done combining two \u sequences.

So maybe not now, but it can be useful later.

Reply:
Full unicode is at most 21 bits, whatever the encoding.  That means just `strtol` is sufficient on any platform.